idea-plugin/src/test/java/com/palantir/javaformat/intellij/PalantirJavaFormatFormattingServiceTest.java

```diff
@@ -0,0 +1,152 @@
+ * limitations under the License.
+ */
+
+package com.palantir.javaformat.intellij;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.intellij.codeInsight.actions.ReformatCodeProcessor;
+import com.intellij.formatting.service.AsyncFormattingRequest;
+import com.intellij.formatting.service.FormattingService;
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.projectRoots.JavaSdk;
+import com.intellij.openapi.projectRoots.Sdk;
+import com.intellij.openapi.roots.ProjectRootManager;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiFile;
+import com.intellij.testFramework.ExtensionTestUtil;
+import com.intellij.testFramework.LightProjectDescriptor;
+import com.intellij.testFramework.fixtures.DefaultLightProjectDescriptor;
+import com.intellij.testFramework.fixtures.IdeaProjectTestFixture;
+import com.intellij.testFramework.fixtures.IdeaTestFixtureFactory;
+import com.intellij.testFramework.fixtures.JavaCodeInsightTestFixture;
+import com.intellij.testFramework.fixtures.JavaTestFixtureFactory;
+import com.intellij.testFramework.fixtures.TestFixtureBuilder;
+import com.palantir.javaformat.intellij.PalantirJavaFormatSettings.State;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PalantirJavaFormatFormattingServiceTest {
+    private JavaCodeInsightTestFixture fixture;
+    private PalantirJavaFormatSettings settings;
+    private DelegatingFormatter delegatingFormatter;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        TestFixtureBuilder<IdeaProjectTestFixture> projectBuilder = IdeaTestFixtureFactory.getFixtureFactory()
+                .createLightFixtureBuilder(getProjectDescriptor(), getClass().getName());
+        fixture = JavaTestFixtureFactory.getFixtureFactory().createCodeInsightFixture(projectBuilder.getFixture());
+        fixture.setUp();
+
+        delegatingFormatter = new DelegatingFormatter();
+        ExtensionTestUtil.maskExtensions(
+                FormattingService.EP_NAME, ImmutableList.of(delegatingFormatter), fixture.getProjectDisposable());
+
+        settings = PalantirJavaFormatSettings.getInstance(fixture.getProject());
+        State resetState = new State();
+        resetState.setEnabled("true");
+        settings.loadState(resetState);
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        fixture.tearDown();
+    }
+
+    @Test
+    public void defaultFormatSettings() throws Exception {
+        String input = Files.readString(
```

Comment:
Deviates from the google-java-format test. This test only checks the `PalantirJavaFormatFormattingService` workflow.